contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -168,7 +168,9 @@ function Repair-AuthorizedKeyPermission
                 $userProfilePath =  $properties.ProfileImagePath
             }
             $userProfilePath = $userProfilePath.Replace("\", "\\")
-            $fullPath -match "^$userProfilePath[\\|\W|\w]+authorized_keys$"
+            if ( $properties.PSChildName -notmatch '\.bak$') {
```

Comment:
It is two separate thing:
- appending a slash to the end of the filesystem *path* in the regex, that's against the value of the registry field "ProfileImagePath" is important to ignore the registry entries that are belonging to users that share the beginning of their path (this is what my other commit is about)
- ignoring the registry *ProfileList* entries that end with .bak are to avoid loading old/backup reg.entries